src/inmanta/agent/executor.py

```diff
@@ -195,10 +195,10 @@ def to_env_blueprint(self) -> EnvBlueprint:
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, ExecutorBlueprint):
             return False
-        return (self.pip_config, set(self.requirements), sorted(list(set(self.sources)))) == (
+        return (self.pip_config, set(self.requirements), self.sources) == (
             other.pip_config,
             set(other.requirements),
```

Comment:
Why keep this `set()` if we already do it in the post init? Why not just `dataclasses.__eq__`?

Reply:
`_hash_cache` would mess it out

Reply:
FYI, `dataclasses.field` has a `compare` boolean: https://docs.python.org/3/library/dataclasses.html#dataclasses.field